File: client/src/game/Side.ts
import { convertColorNameToSideColor, SIDE_COLOR } from "@/utils/colors";

interface ISide {
  id: string;
  name: string;
  totalScore?: number;
  color?: string | SIDE_COLOR;
}

export default class Side {
  id: string;
  name: string;
  totalScore: number;
  color: SIDE_COLOR;

  constructor(parameters: ISide) {
    this.id = parameters.id;
    this.name = parameters.name;
    this.totalScore = parameters.totalScore ?? 0;
    this.color = convertColorNameToSideColor(parameters.color);
  }
}

File: gym/blade/Side.py
import json
from blade.utils.colors import convert_color_name_to_side_color, SIDE_COLOR


class Side:

    def __init__(
        self,
        id: str,
        name: str,
        total_score: int = 0,
        color: str | SIDE_COLOR | None = None,
    ):
        self.id = id
        self.name = name
        self.total_score = total_score
        self.color = convert_color_name_to_side_color(color)

    def to_dict(self):
        return {
            "id": str(self.id),
            "name": self.name,
            "total_score": self.total_score,
            "color": (
                self.color.value if isinstance(self.color, SIDE_COLOR) else self.color
            ),
        }
